contracts/Core.sol

```diff
@@ -22,60 +22,112 @@ pragma solidity ^0.4.23;
 // ----------------------------------------------------------------------------
 
 import "./CoreInterface.sol";
+import "./MerklePatriciaProof.sol";
+import "./util.sol";
+import "./WorkersInterface.sol";
+import "./RLP.sol";
 
 
-/// @dev Core is a minimal stub that will become the anchoring and consensus point for
-///      the utility chain to validate itself against
-contract Core is CoreInterface {
+/**
+*	@dev Core is a minimal stub that will become the anchoring and consensus point for
+*      the utility chain to validate itself against
+*/
+contract Core is CoreInterface, Util {
 
-	/*
-	 *  Structures
-	 */
-	// struct stakeTokenTuple {
-	// 	address stake;
-	// 	address token;
-	// }
+	/**
+	*    Events
+	*/
+
+	/** event StateRootCommitted on successful execution of commitStateRoot */
+	event StateRootCommitted(uint256 blockHeight, bytes32 stateRoot);
 
+	/** event OpenSTProven on successful execution of proveOpenST */
+	event OpenSTProven(uint256 blockHeight, bytes32 storageRoot, bytes32 hashedAccount);
 
-	/*
+	/**
 	 *  Storage
 	 */
-	/// registrar registers for the two chains
-	address private coreRegistrar;
-	/// chainIdOrigin stores the chainId this chain
-	uint256 private coreChainIdOrigin;
-	/// chainIdRemote stores the chainId of the remote chain
+
+	/** Mapping of block height to state root of the block.  */
+	mapping (uint256 /* block height */ => bytes32) private stateRoots;
+
+	/** Mapping of block height to storafe root of the block.  */
```

Comment:
Hi @abhayks1, looking into this PR out of interest, 
I noticed that the comment `storage root of the block` has a typo.

Reply:
thanks. taken care.